### PR TITLE
fix: state and client nonsense

### LINKED ISSUE #26: [UI] Console doesn't get updated
- run the tests so they fail
- open the console, see errors, don't close/hide it
- make it so different tests fail - or none fail?
- run the tests again
- console doesn't get updated until you click tests and then go back to the console

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #26

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
@moT01 Well... this ended up being more than I expected. I completely refactored the client. It looks nicer (file-system and code) though, and that bug is fixed.

**Comment by moT01** [comment]
Seeing some issues on this one - blocked for now. Will explain in upcoming meeting.

File: .freeCodeCamp/client/index.tsx [hunk 1]
```diff
@@ -1,14 +1,161 @@
...
+  const [tests, setTests] = useState<TestType[]>([]);
+  const [hints, setHints] = useState('');
+  const [cons, setCons] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [alertCamper, setAlertCamper] = useState<null | string>(null);
+
+  useEffect(() => {
+    socket.onopen = function (_event) {
+      sock(Events.CONNECT);
+      window.location.reload();
```
Comment (ShaunSHamilton) [inline .freeCodeCamp/client/index.tsx]: Does not seem to help anyway.

```suggestion
```

Comment (ShaunSHamilton) [inline .freeCodeCamp/client/index.tsx]:
```suggestion

```

**Pull request merged**